spec/controllers/alchemy/admin/pages_controller_spec.rb

```diff
@@ -118,32 +118,59 @@ module Alchemy
       end
 
       describe "#flush" do
-        let(:content_page_1) { create(:alchemy_page, :public, name: "content page 1", published_at: Time.current - 5.days) }
-        let(:content_page_2) { create(:alchemy_page, :public, name: "content page 2", published_at: Time.current - 8.days) }
-        let(:layout_page_1)  { create(:alchemy_page, layoutpage: true, name: "layout_page 1", published_at: Time.current - 5.days) }
-        let(:layout_page_2)  { create(:alchemy_page, layoutpage: true, name: "layout_page 2", published_at: Time.current - 8.days) }
-        let(:content_pages)  { [content_page_1, content_page_2] }
-        let(:layout_pages)   { [layout_page_1, layout_page_2] }
+        let(:content_page_1) do
+          time = Time.current - 5.days
+          create :alchemy_page,
+            public_on: time,
+            name: "content page 1",
+            published_at: time
+        end
 
-        before do
-          content_pages
-          layout_pages
+        let(:content_page_2) do
+          time = Time.current - 8.days
+          create :alchemy_page,
+            public_on: time,
+            name: "content page 2",
+            published_at: time
+        end
+
+        let(:layout_page_1) do
+          create :alchemy_page,
+            layoutpage: true,
+            name: "layout_page 1",
+            published_at: Time.current - 5.days
+        end
+
+        let(:layout_page_2) do
+          create :alchemy_page,
+            layoutpage: true,
+            name: "layout_page 2",
+            published_at: Time.current - 8.days
         end
 
+        let(:content_pages) { [content_page_1, content_page_2] }
```

Comment:
Could you have used `let!` here instead of calling `content_pages` in the `it` block?